cosmo_tester/conf_cli.py

```diff
@@ -0,0 +1,156 @@
+    if platform == 'openstack':
+        config.raw_config['openstack'] = {}
+        target = config.raw_config['openstack']
+        target['username'] = os.environ["OS_USERNAME"]
+        target['password'] = os.environ["OS_PASSWORD"]
+        target['tenant'] = (
+            os.environ.get("OS_TENANT_NAME")
+            or os.environ['OS_PROJECT_NAME']
+        )
+        target['url'] = os.environ["OS_AUTH_URL"]
+        target['region'] = os.environ.get("OS_REGION_NAME", "RegionOne")
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description=(
+            'Tool for working with test configs.'
+        ),
+    )
+
+    subparsers = parser.add_subparsers(help='Action',
+                                       dest='action')
+
+    validate_args = subparsers.add_parser('validate',
+                                          help='Validate the config.')
+    validate_args.add_argument(
+        '-c', '--config-location',
+        help='The config file to validate.',
+        default='test_config.yaml',
+    )
+
+    subparsers.add_parser('schema',
+                          help='Print the schema.')
+
+    generate_args = subparsers.add_parser('generate',
+                                          help='Generate sample config.')
+    generate_args.add_argument(
+        '-i', '--include-defaults',
+        help=(
+            'Include entries with default values. '
+            'This will result in a much longer config, most of which will '
+            'not be required, but will allow for easy modification of any '
+            'settings.'
+        ),
+        action='store_true',
+        default=False
+    )
+    generate_args.add_argument(
+        '-p', '--platform',
+        help=(
+            'Generate the initial config including values for the specified '
+            'target platform. For platform configuration to be properly '
+            "generated, you should have that platform's default way of "
+            'authenticating prepared, e.g. ". openstackrc".'
+        ),
+    )
+
+    args = parser.parse_args()
+
+    logger = get_logger('conf_cli')
+
+    if args.action == 'validate':
+        # We validate on loading, so simply attempting to load the config with
+        # the config supplied by the user will validate it
+        load_config(logger, args.config_location)
+    elif args.action == 'schema':
+        config = load_config(logger, validate=False)
+        print(show_schema(config.schema, False))
+    elif args.action == 'generate':
+        config = load_config(logger, validate=False)
+        if args.platform:
+            apply_platform_config(logger, config, args.platform)
+            if not config.check_config_is_valid(fail_on_missing=False):
+                sys.exit(1)
+        print(show_schema(config.schema,
+                          generate_sample_config=True,
+                          include_defaults=args.include_defaults,
+                          raw_config=config.raw_config))
```

Comment:
@geokala 
```suggestion
                          raw_config=config.raw_config))


if __name__ == '__main__':
    main()
```
if you do this, then it will be possible to run this via `python -m cosmo_tester.conf_cli`, which matters
1) from an editable installation
2) if for whatever reason you don't want to touch your PATH
3) it will just happen to work without even installing it, because it just happens to have no non-stdlib deps

Reply:
Added.